src/ccl_power.c

```diff
@@ -1358,47 +1289,46 @@ static void ccl_cosmology_compute_power_emu(ccl_cosmology * cosmo, int * status)
       xstar[6] = cosmo->params.wa;
       if ((cosmo->params.N_nu_mass>0) && (cosmo->config.emulator_neutrinos_method == ccl_emu_equalize)){
 	xstar[7] = Omeganuh2_eq;
-      } else { 
+      }else{
         xstar[7] = cosmo->params.Omega_n_mass*cosmo->params.h*cosmo->params.h;
       }
-      xstar[8] = 1./aemu[j]-1;
+      xstar[8] = 1./zemu[j]-1;
       //Need to have this here because otherwise overwritten by emu in each loop
-
+      
       //Call emulator at this redshift
-      ccl_pkemu(xstar,&y, status, cosmo);
-      if (y == NULL){
-	*status = CCL_ERROR_MEMORY;
-	ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_emu(): Error obtaining emulator predictions\n");
+      ccl_pkemu(xstar,NK_EMU,y, status, cosmo);
+      ccl_check_status(cosmo, status);
+      if (y == NULL) {
+	*status=CCL_ERROR_MEMORY;
+	ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_emu(): memory allocation error\n");
+	break;
       }
       for (int i=0; i<NK_EMU; i++){
 	logx[i] = log(mode[i]);
```

Comment:
I cannot find where `mode` is defined. Is this a global var? Does it need a more descriptive name?

Reply:
`include/ccl_emu17_params.h`. It is a global variable indeed (which hasn't been added in this PR)